### DISCUSSION TITLE
Dose the output of ecapa-tdnn in the classification task is probability?

### DISCUSSION THREAD

**Discussion by lhanzl** [speechbrain/speechbrain#1628] [Q&A]
Hi, I am using ecapa-tdnn for classification tasks and found that when making predictions, the difference in the scores of the output nodes is very small. details as follows:
```
output                      softmax(output)
[-0.2018069326877594,	[0.01732958983316111,
 -0.14673146605491638,	 0.01831079729647824,
 -0.14109285175800323,	 0.018414336454263124,
 -0.11534786224365234,	 0.018894568614446584,
 -0.0882330909371376,	 0.019413899472335516,
 -0.07039536535739899,	 0.019763306330780574,
 -0.06926165521144867,	 0.01978572489736229,
 -0.06246980279684067,	 0.019920564006109087,
 -0.061220571398735046,	 0.019945464950415902,
 -0.055079925805330276,	 0.020068319799871236,
 -0.05080120638012886,	 0.020154370471690046,
 -0.04340587928891182,	 0.020303971124711728,
 -0.03528368100523949,	 0.0204695555485935,
 -0.03400317206978798,	 0.020495783786540846,
 -0.028194941580295563,	 0.02061517441132731,
 -0.018447529524564743,	 0.020817101545616938,
 -0.018377292901277542,	 0.02081856371988462,
 -0.017970150336623192,	 0.020827041569039466,
 -0.016880501061677933,	 0.02084974810861993,
 -0.005735216662287712,	 0.021083424255622745,
 -0.003223307430744171,	 0.021136450474285508,
 -0.0030825547873973846,	 0.021139425694940907,
 -0.0021692796144634485,	 0.02115874062617926,
 -0.002124277874827385,	 0.02115969282774115,
 -0.0006143711507320404,	 0.02119166612249347,
 0.007602931000292301,	 0.021366521883503107,
 0.009185916744172573,	 0.021400371567756298,
 0.014768049120903015,	 0.021520165315972672,
 0.021696316078305244,	 0.021669780454588672,
 0.03147256746888161,	 0.021882668604362134,
 0.041788939386606216,	 0.022109586826096966,
 0.05316564068198204,	 0.022362557247163326,
 0.0678737685084343,	 0.02269389933535364,
 0.06846173107624054,	 0.022707246422089582,
 0.06926628947257996,	 0.02272552307918929,
 0.08045008778572083,	 0.02298110728298396,
 0.11100300401449203,	 0.023694083417103193,
 0.17250362038612366,	 0.025197026423772348,
 0.17413178086280823,	 0.02523808464193942,
 0.18498116731643677,	 0.025513393139186536,
 0.21954409778118134,	 0.026410626969863474,
 0.2639785408973694,	 0.027610631808721708,
 0.3827281892299652,	 0.03109200018574848,
 0.3887121379375458,	 0.03127861089769992,
 0.5425097942352295]	 0.03647883255439419]
```
The above output comes from the officially released language recognition model.
```
output
[0.9950, 0.9418]
---
softmax(output)
[0.51330666 0.48669334]
```
The above output is a gender classification model trained by myself, using CommonLanguage recipe.
The score difference of the output nodes is very small, especially in the two-class model. But my gender classification model has a good accuracy rate.
1. Does the output of ecapa-tdnn in the classification task become probabilities after softmax?
2. I want to get a reasonable degree of confidence, could you give me some suggestions.
Thank you very much.

**Comment by mravanelli** [comment]
Hi @lhanzl,
if you use the default AAM softmax  is it normal to have output scores like that. You can replace it with a standard softmax to have standard posterior probabilities. Normally, AAM softmax provides better speaker embeddings (not sure if it is the best solution for gender classification, but you can try and compare the two)

**Comment by jakubbortlik** [comment]
Hi @mravanelli, I have an additional question about this topic. lhanzl gives the example with output values ranging from "-0.2018069326877594" to "0.5425097942352295". The SpeechBrain [documentation](https://speechbrain.readthedocs.io/en/latest/API/speechbrain.pretrained.interfaces.html#speechbrain.pretrained.interfaces.EncoderClassifier.classify_file) says that the output of the "classify_file" method is the "log posterior probabilities of each class ([batch, N_class])" and that the "score" "is the value of the log-posterior for the best class ([batch,])". However, as far as I know, log values that correspond to posterior probabilities should only have values smaller or equal to 0. Am I missing something here, or is there just a mistake in the documentation? What sort of thing are really the out_prob and score outputs of classify_files?
Thanks!
